atc/worker/worker.go

```diff
@@ -583,9 +605,11 @@ func (worker *gardenWorker) cloneRemoteVolumes(
 		}
 
 		g.Go(func() error {
-			err = nonLocalInput.desiredArtifact.StreamTo(groupCtx, logger.Session("stream-to", destData), inputVolume)
-			if err != nil {
-				return err
+			if streamable, ok := nonLocalInput.desiredArtifact.(StreamableArtifactSource); ok {
```

Comment:
Hey, 

was it intended to ignore the case where `nonLocalInput.desiredArtifact`'s type value does not match `StreamableArtifactSource`? i.e., do we want to proceed if that's not the case?

Just wondering if that's an "impossible statement" that we should treat as a programming mistake.

thanks!

Reply:
its an impossible case, the only types going in there should be of Streamable type. Its something that is hopefully temporary and gets fixed.

Reply:
should we `panic`?

https://github.com/concourse/concourse/blob/cfe2a7ae78655c2e36f7a3afddb71961e0899ab5/atc/plan_factory.go#L65